src/renderer/views/Popular/Popular.vue

```diff
@@ -22,5 +22,102 @@
   </div>
 </template>
 
-<script src="./Popular.js" />
+<script setup>
+
+import { computed, onBeforeUnmount, onMounted, ref } from 'vue'
+
+import FtLoader from '../../components/ft-loader/ft-loader.vue'
+import FtCard from '../../components/ft-card/ft-card.vue'
+import FtElementList from '../../components/FtElementList/FtElementList.vue'
+import FtRefreshWidget from '../../components/ft-refresh-widget/ft-refresh-widget.vue'
+import store from '../../store/index'
+
+import { invidiousAPICall } from '../../helpers/api/invidious'
+import { copyToClipboard, getRelativeTimeFromDate, setPublishedTimestampsInvidious, showToast } from '../../helpers/utils'
+import { useI18n } from '../../composables/use-i18n-polyfill'
+
+const { t } = useI18n()
+
+const isLoading = ref(false)
+const shownResults = ref([])
```

Comment:
As we replace the entire array but don't modify the arrays themselves, we can use `shallowRef` here, as we only need to know when the arrays is replaced (`.value = ...`).

And yes before you mention it I am aware than the moment we write the array to the store, VueX will make the entire thing deeply reactive anyway, but once we switch to Pinia we can define the popularCache as `shallowReactive` too. If you prefer we can also leave it for the moment and only update it after or while we switch to Pinia?
```suggestion
const shownResults = shallowRef([])
```